dv/uvm/core_ibex/Makefile

```diff
@@ -414,30 +414,42 @@ wave-arg := $(if $(call equal,$(WAVES),1),--en_wave,)
 cosim-arg := $(if $(call equal,$(COSIM),1),--en_cosim,)
 lsf-arg := $(if $(LSF_CMD),--lsf_cmd="$(LSF_CMD)",)
 
-# Extract LDFLAGS and "LIBS" for the vcs build from pkg-config. VCS expects
-# most things to be supplied with -LDFLAGS arguments, but unfortunately these
-# come before the object files that are being linked. To add libraries
-# themselves, you're supposed to pass them directly (-lfoo). Split things out
-# here to make sense of them.
-SPIKE_PCS      := riscv-riscv riscv-disasm riscv-fdt
-SPIKE_CFLAGS   := $(if $(cosim-arg),$(shell pkg-config --cflags $(SPIKE_PCS) || echo FAIL),)
-SPIKE_ALL_LIBS := $(if $(cosim-arg),$(shell pkg-config --libs $(SPIKE_PCS) || echo FAIL),)
-SPIKE_LIBS     := $(filter -l%,$(SPIKE_ALL_LIBS))
-SPIKE_LDFLAGS  := $(filter-out -l%,$(SPIKE_ALL_LIBS))
-check-spike-pkgconfig := \
-  if [ x'$(SPIKE_CFLAGS)' == xFAIL -o x'$(SPIKE_ALL_LIBS)' == xFAIL ]; then \
-	  echo >&2 "Failed to find Spike pkg-config packages."; \
-      exit 1; \
-  fi
+# We need to the different simulators to link against the built ISS correctly
+# for cosimulation. This requires some finesse-ing.
+# Extract CFLAGS,LDFLAGS and LIBS from pkg-config. Allow the simulator yaml file
+# to format them appropriately for each simulator.
+# - VCS expects most things to be supplied with -LDFLAGS arguments, but
+# unfortunately these come before the object files that are being linked. To add
+# libraries themselves, you're supposed to pass them directly (-lfoo). Split
+# things out here to make sense of them.
+# - Xcelium expects CFLAGS, includes, and libs to be passed directly to xrun.
+# However, linker flags are to be passed using -Wld. This can be finnicky, so
+# using the alternate -Xlinker style simplifies things.
+SPIKE_PCS          := riscv-riscv riscv-disasm riscv-fdt
+ifeq ($(COSIM),1)
```

Comment:
Eek! This is getting a bit crazy! If I'm right and this is only needed when compiling the testbench, I wonder whether it would be better to move this into the `sim.py` wrapper? It might be a bit less horrible than expressing everything in Makefile syntax.

Reply:
I agree!! The Makefile level is very tricky for these sorts of ops.
Can I suggest merging as-is then migrating this logic to python-land in a follow-up PR?